handler.go

```diff
@@ -198,11 +202,21 @@ func (h *Handler) rateLimitExceeded(w http.ResponseWriter, r *http.Request, repl
 	if err != nil {
 		remoteIP = r.RemoteAddr // assume there was no port, I guess
 	}
-	h.logger.Info("rate limit exceeded",
-		zap.String("zone", zoneName),
-		zap.Duration("wait", wait),
-		zap.String("remote_ip", remoteIP),
-	)
+
+	if h.LogKey {
+		h.logger.Info("rate limit exceeded",
+			zap.String("zone", zoneName),
+			zap.String("key", key),
+			zap.Duration("wait", wait),
+			zap.String("remote_ip", remoteIP),
+		)
+	} else {
+		h.logger.Info("rate limit exceeded",
+			zap.String("zone", zoneName),
+			zap.Duration("wait", wait),
+			zap.String("remote_ip", remoteIP),
+		)
+	}
```

Comment:
What you can do here instead to avoid duplication is `logger := h.logger.With(...)` and put all the basic fields in there. Then, do `if h.LogKey { ... }` where you can add the key like `logger = h.With(zap.String("key"))`.

You could alternatively build a slice of zap fields instead of using `logger.With()`.

Reply:
Didn't know about that. Should be fixed now.